minitrace/src/collector/global_collector.rs

```diff
@@ -31,31 +34,54 @@ thread_local! {
     };
 }
 
-pub(crate) fn start_collect(collect_args: CollectArgs) -> u32 {
-    let collect_id = NEXT_COLLECT_ID.fetch_add(1, Ordering::AcqRel);
-    send_command(CollectCommand::StartCollect(StartCollect {
-        collect_id,
-        collect_args,
-    }));
-    collect_id
-}
+#[derive(Default, Clone, Debug)]
+pub(crate) struct GlobalCollect;
 
-pub(crate) fn drop_collect(collect_id: u32) {
-    force_send_command(CollectCommand::DropCollect(DropCollect { collect_id }));
-}
+#[mockall::automock]
+impl GlobalCollect {
+    pub fn start_collect(&self, collect_args: CollectArgs) -> u32 {
+        let collect_id = NEXT_COLLECT_ID.fetch_add(1, Ordering::AcqRel);
+        send_command(CollectCommand::StartCollect(StartCollect {
+            collect_id,
+            collect_args,
+        }));
+        collect_id
+    }
 
-pub(crate) fn commit_collect(
-    collect_id: u32,
-    tx: futures::channel::oneshot::Sender<Vec<SpanRecord>>,
-) {
-    force_send_command(CollectCommand::CommitCollect(CommitCollect {
-        collect_id,
-        tx,
-    }));
-}
+    pub async fn commit_collect(&self, collect_id: u32) -> Vec<SpanRecord> {
+        let (tx, rx) = futures::channel::oneshot::channel();
+        force_send_command(CollectCommand::CommitCollect(CommitCollect {
+            collect_id,
+            tx,
+        }));
+        rx.await.unwrap_or_else(|_| Vec::new())
+    }
 
-pub(crate) fn submit_spans(spans: SpanSet, parents: ParentSpans) {
-    send_command(CollectCommand::SubmitSpans(SubmitSpans { spans, parents }));
+    pub fn drop_collect(&self, collect_id: u32) {
+        force_send_command(CollectCommand::DropCollect(DropCollect { collect_id }));
+    }
+
+    /// Note that: relationships are not built completely here, and a further job is needed.
+    ///
+    /// Every `SpanSet` has its own root span**s** whose `raw_span.parent_id`s are equal to
+    /// `SpanId::default()`, and such a root span can have multiple parents where mainly comes
+    /// from [`Span::enter_with_parents`].
+    ///
+    /// Parents of root spans are recorded into `CollectToken` which has several `CollectTokenItem`s.
+    /// Look into a `CollectTokenItem`, `parent_id_of_roots` is found.
+    ///
+    /// So the expected further job mentioned above is:
+    /// * Copy `SpanSet` to the same number of copies as `CollectTokenItem`s, one `SpanSet` to one
+    ///   `CollectTokenItem`
+    /// * Amend `raw_span.parent_id` of root spans in `SpanSet` to `parent_id_of_roots` of `CollectTokenItem`
+    ///
+    /// [`Span::enter_with_parents`]: crate::Span::enter_with_parents
+    pub fn submit_spans(&self, spans: SpanSet, collect_token: CollectToken) {
+        send_command(CollectCommand::SubmitSpans(SubmitSpans {
+            spans,
+            collect_token,
+        }));
+    }
 }
 
 fn send_command(cmd: CollectCommand) {
```

Comment:
Move this above so that it will directly follwing the definition of `COMMAND_SENDER`